src/base-element.js

```diff
@@ -589,40 +591,101 @@ export class BaseElement {
+        }
+      }
+      return true;
+    }, 'There is already a default action registered.');
+    this.actionMap_[method] = {handler, minTrust, isDefault: opt_isDefault};
+  }
+
+  /**
+   * Registers the default action for this component.
+   * @param {function(!./service/action-impl.ActionInvocation)} handler
+   * @param {string=} methodName
+   */
+  registerDefaultAction(handler, methodName = DEFAULT_METHOD_) {
+    this.registerAction(
+        methodName, handler, ActionTrust.HIGH, /* opt_isDefault */ true);
   }
 
   /**
    * Requests the element to execute the specified method. If method must have
    * been previously registered using {@link registerAction}, otherwise an
    * error is thrown.
    * @param {!./service/action-impl.ActionInvocation} invocation The invocation data.
-   * @param {boolean} unusedDeferred Whether the invocation has had to wait any time
+   * @param {?Array<{tagOrTarget: !Element, method: string}>} opt_whitelist
+   *     Actions permitted for execution.
+   * @param {boolean=} unusedDeferred Whether the invocation has had to wait any time
    *   for the element to be resolved, upgraded and built.
    * @final
    * @package
    */
-  executeAction(invocation, unusedDeferred) {
-    if (invocation.method == 'activate') {
+  executeAction(invocation, opt_whitelist = [], unusedDeferred) {
+    const whitelist = opt_whitelist || [];
+    if (invocation.method == DEFAULT_METHOD_) {
       if (invocation.satisfiesTrust(this.activationTrust())) {
-        return this.activate(invocation);
+        for (const actionName in this.actionMap_) {
+          const action = this.actionMap_[actionName];
+          // Check if the action is whitelisted. Note that we pass in
+          // the actual method/action name and not 'activate', which is the
+          // default name/method, so that we don't have to whitelist
+          // 'activate', as it is implied.
+          const actionWhitelisted = this.isActionWhitelisted_(
+              invocation, actionName, whitelist);
+          const {method, tagOrTarget} = invocation;
+          if (!actionWhitelisted) {
+            user().error(TAG_,
+                `"${tagOrTarget}.${method}" is not whitelisted ` +
+                `whitelist: ${JSON.stringify(whitelist)}.`);
+          }
+          if (action.isDefault && actionWhitelisted) {
+            return action.handler(invocation);
+          }
+        }
       }
     } else {
       this.initActionMap_();
-      const holder = this.actionMap_[invocation.method];
-      user().assert(holder, `Method not found: ${invocation.method} in %s`,
+      const {method} = invocation;
+      const holder = this.actionMap_[method];
+      user().assert(holder, `Method not found: ${method} in %s`,
           this);
       const {handler, minTrust} = holder;
-      if (invocation.satisfiesTrust(minTrust)) {
+      if (invocation.satisfiesTrust(minTrust)
+          && this.isActionWhitelisted_(invocation, method, whitelist)) {
```

Comment:
It might be cleaner if we refactor to only call `satisfiesTrust` and `isActionWhitelisted_` once in this function.